Add liberty tools eclipse blog (#2405)
OpenLiberty/blogs: posts/2022-08-01-liberty-tools-eclipse.adoc

---
layout: post
title: "Introducing Liberty Tools for the Eclipse IDE"
categories: blog
author_picture: https://avatars3.githubusercontent.com/awisniew90
author_github: https://github.com/awisniew90
seo-title: Introducing Liberty Tools for the Eclipse IDE
seo-description: A next-generation plugin that brings the power of Open Liberty into your favorite IDE
blog_description: "A next-generation plugin that brings the power of Open Liberty into your favorite IDE"
open-graph-image: https://openliberty.io/img/blog/liberty-tools-actions.png
---
= Introducing Liberty Tools for the Eclipse IDE
Adam Wisniewski <https://github.com/awisniew90>

> Liberty Tools for Eclipse provides a new experience for developing applications with Open Liberty. This early release supports editing Liberty server.xml configuration, developing MicroProfile applications, a Liberty Dashboard for organizing your projects, and Liberty dev mode functions all from within your IDE. 

Reviewer: ```suggestion
Liberty Tools for Eclipse provides a new experience for developing applications with Open Liberty. This early release supports editing Liberty `server.xml` configuration, developing MicroProfile applications, a Liberty Dashboard for organizing your projects, and Liberty dev mode functions, all from within your IDE. 
```

> Check out a quick demo:

Reviewer: ```suggestion
Check out this quick video demo:
```

++++
<iframe width="560" height="315" align="center" src="https://www.youtube.com/embed/_ucSs20sUVc" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
++++

== Liberty Dashboard

> The Liberty Dashboard keeps all of your Maven and Gradle projects organized and is automatically populated with projects that are configured to run with Open Liberty. With a few clicks, you can start and stop your app, run tests and view test reports. 

Reviewer: ```suggestion
The Liberty Dashboard keeps all of your Maven and Gradle projects organized and is automatically populated with projects that are configured to run with Open Liberty. With a few clicks, you can start and stop your app, run tests, and view test reports. 
```

> == Dev Mode

Reviewer: ```suggestion
== Dev mode
```

> Liberty Dev mode automatically picks up your code changes and deploys them to your running app all without having to restart the server. With Liberty Tools, your app starts in dev mode automatically giving you all of the benefits of faster development. 

Reviewer: ```suggestion
link:/docs/latest/development-mode.html[Liberty dev mode] automatically picks up your code changes and deploys them to your running app, without having to restart the server. With Liberty Tools, your app starts in dev mode automatically, giving you all of the benefits of faster development. 
```

== MicroProfile and Liberty config support

> Configuring your Liberty server is now simpler than ever with server config auto-complete, validation, and "hover over" descriptions.

Reviewer: ```suggestion
Configuring your Liberty server is now simpler than ever with server configuration auto-complete, validation, and hover-over descriptions in your IDE.
```

> Additionally, you can quickly develop MicroProfile applications with automatic code blocks, auto-complete, and real time MP syntax validation. 

Reviewer: ```suggestion
Also, you can quickly develop MicroProfile applications with automatic code blocks, auto-complete, and real-time MicroProfile syntax validation. 
```

== Try it out

> === Installation

Reviewer: ```suggestion
=== Installing Liberty Tools for Eclipse
```

> Install the plugin using the instructions found link:https://github.com/OpenLiberty/liberty-tools-eclipse/blob/main/docs/installation/installation.md[here]

Reviewer: ```suggestion
For installation instructions, see link:https://github.com/OpenLiberty/liberty-tools-eclipse/blob/main/docs/installation/installation.md[Liberty Tools for Eclipse].
```

=== Using the Dashboard

> Once installed, you will see the Open Liberty UFO logo in the top tray. Click on that to open the Liberty Dashboard tab:

Reviewer: ```suggestion
After you install the plug-in, look for the Open Liberty UFO logo in the menu. Click the icon to open the Liberty Dashboard tab:
```

[.img_border_light]
image::/img/blog/liberty-tools-dash-icon.png[Liberty Dashboard Icon,width=70%,align="center"]

> The Dashboard will automatically populate with any Maven and Gradle projects in your workspace that are configured to run with Open Liberty (server.xml in the default location). If you do not have any apps in your current workspace, try importing one like the Open Liberty "Getting Started" app link:https://github.com/openliberty/guide-getting-started[here]. Once imported, refresh the Dashboard tab by clicking the refresh icon in the top right. 

Reviewer: The Dashboard automatically populates with any Maven and Gradle projects in your workspace that are configured to run with Open Liberty, with the `server.xml ` file in the default location. If you do not have any apps in your current workspace, try importing one, like the link:https://github.com/openliberty/guide-getting-started[Open Liberty Starter app]. Once you import an app, refresh the Dashboard tab by clicking the refresh icon.

image::/img/blog/liberty-tools-terminal-start.png[Start in Dev Mode,width=70%,align="center"]

Once the app is running, right-click on the app in the dashboard again and select **Run tests**. 

[.img_border_light]
image::/img/blog/liberty-tools-test-action.png[Run Tests,width=70%,align="center"]

Any automated tests associated with the app are run:

[.img_border_light]
image::/img/blog/liberty-tools-terminal-test.png[Tests Complete,width=70%,align="center"]

=== Liberty Config Support

Open `src/main/liberty/config/server.xml`. Start adding a new feature to the Feature Manager list. You will see a Content Assist pop up menu with auto-complete options:

[.img_border_light]
image::/img/blog/liberty-tools-feature-autocomplete.png[Feature Auto-complete,width=70%,align="center"]

To get a list of available features and their descriptions, engage Content Assist by entering `Ctrl+Space` from within the feature configuration element:

[.img_border_light]
image::/img/blog/liberty-tools-feature-list.png[Feature List,width=70%,align="center"]

Similar capabilities are available for many other Liberty config elements. Configuring your server is simpler than ever before. 

== Learn more

> https://github.com/OpenLiberty/liberty-tools-eclipse

Reviewer: ```suggestion
link:https://github.com/OpenLiberty/liberty-tools-eclipse[Liberty Tools for Eclipse]
```
